lib/services/sshd.pm

```diff
@@ -23,4 +32,168 @@ sub check_sshd_service {
+    assert_script_run 'until ss -tulpn|grep sshd|egrep "4242|5252";do sleep 1;done';
+
+    # Scan public keys on forwarded ports
+    # Add a workaround about bsc#1193275 in FIPS test
+    my $output_4242 = script_output("ssh-keyscan -p 4242 localhost >> ~/.ssh/known_hosts", proceed_on_failure => 1);
+    if ($output_4242 =~ /choose_kex: unsupported KEX method curve25519-sha256/) {
+        record_soft_failure('bsc#1193275', "Currently the curve25519 is not FIPS approved");
+    }
+
+    my $output_5252 = script_output("ssh-keyscan -p 5252 localhost >> ~/.ssh/known_hosts", proceed_on_failure => 1);
+    if ($output_5252 =~ /choose_kex: unsupported KEX method curve25519-sha256/) {
+        record_soft_failure('bsc#1193275', "Currently the curve25519 is not FIPS approved");
+    }
+
+    # Connect to forwarded ports
+    assert_script_run "ssh -v -p 4242 $ssh_testman\@localhost whoami";
+    assert_script_run "ssh -v -p 5252 $ssh_testman\@localhost whoami";
+
+    # Copy the list of known hosts to $ssh_testman's .ssh directory
+    assert_script_run "install -m 0400 -o $ssh_testman ~/.ssh/known_hosts /home/$ssh_testman/.ssh/known_hosts";
+
+    # Test SSH command within SSH command
+    assert_script_run "ssh -v -p 4242 -tt $ssh_testman\@localhost ssh -tt $ssh_testman\@localhost whoami";
+
+    # Test ProxyCommand option
+    assert_script_run "ssh -v -t -o ProxyCommand='ssh -v $ssh_testman\@localhost nc localhost 4242' $ssh_testman\@localhost whoami";
+
+    # Test JumpHost option
+    if (is_leap('15.0+') || is_tumbleweed || is_sle('15+')) {
+        assert_script_run("ssh -v -J $ssh_testman\@localhost:4242 $ssh_testman\@localhost whoami");
+    }
+
+    # SCP (poo#46937)
+    assert_script_run "echo 'sshd.pm: Testing SCP subsystem' | logger";
+    assert_script_run "scp -4v $ssh_testman\@localhost:/etc/resolv.conf /tmp";
+    assert_script_run "scp -4v '$ssh_testman\@localhost:/etc/{group,passwd}' /tmp";
+    assert_script_run "scp -4v '$ssh_testman\@localhost:/etc/ssh/*.pub' /tmp";
+}
+
+sub do_ssh_cleanup {
+    # Restore ~/.ssh generated in consotest_setup
+    # poo#68200. Confirm the ~/.ssh_bck directory is exist in advance and then restore, in order to avoid the null restore
+    assert_script_run 'rm -rf ~/.ssh';
+    assert_script_run 'if [ -d ~/.ssh_bck ]; then mv ~/.ssh_bck ~/.ssh; fi';
+
+    # Restore the /etc/ssh/sshd_config
+    assert_script_run 'cp /etc/ssh/sshd_config{_before,}';
+
+    # Kill $ssh_testman to stop all SSH sessions
+    assert_script_run "killall -u $ssh_testman || true";
+    wait_still_screen 3;
+
+    record_info("Restart sshd", "Restart sshd.service");
+    systemctl("restart sshd");
+
+    # Clear the remains from background commands
+    clear_console if !is_serial_terminal;
+}
+
+sub check_function {
+    ssh_basic_check();
+
+    wait_still_screen 3;
+    systemctl("restart sshd");
+}
+
+# check sshd service
+sub full_sshd_check {
+    configure_service();
```

Comment:
Why this is not considering the migration phase?
Does that mean that we configure the service before as well as after migration? Why?

Reply:
yes, cause we migration need to reboot and we've backed the ssh key and user sshboy, we need to clean them up before migration.  or it will block the migration.
https://openqa.nue.suse.com/tests/7960464#step/first_boot/5